podcasts/Sharing/Clip/VideoExporter.swift

```diff
@@ -0,0 +1,203 @@
+import SwiftUI
+import AVFoundation
+import UIKit
+import PocketCastsUtils
+
+protocol AnimatableContent: View {
+    func update(for progress: Double)
+}
+
+@available(iOS 16, *)
+class VideoExporter {
+
+    struct Parameters {
+        let duration: TimeInterval
+        let size: CGSize
+        let fps: Int = 60
+        let audioPlayerItem: AVPlayerItem
+        let audioStartTime: CMTime
+        let audioDuration: CMTime
+        let additionalLoadingCount: Int64 = 50
+        let fileType: AVFileType
+    }
+
+    enum ExportError: Error {
+        case failedToCreateCompositionTrack
+        case failedToCreateExportSession
+        case failedToAddAudioTrack
+        case exportFailed(Error?)
+    }
+
+    @MainActor
+    static func export<Content: AnimatableContent>(view: Content, with parameters: Parameters, to outputURL: URL, progress: Progress) async throws {
+        let loopDuration: Double = 5
+        let loopFrameCount = Int(loopDuration * Double(parameters.fps))
+
+        let start = Date()
+        FileLog.shared.addMessage("VideoExporter Started: \(start)")
+
+        let temporaryFileURL = FileManager.default.temporaryDirectory
+            .appendingPathComponent(UUID().uuidString)
+            .appendingPathExtension(UTType(parameters.fileType.rawValue)?.preferredFilenameExtension ?? ".mp4")
+
+        progress.totalUnitCount = Int64(loopFrameCount) + parameters.additionalLoadingCount
+
+        try await withTaskCancellationHandler {
+            // Export initial seconds long video
+            try await exportInitialVideo(of: view, to: temporaryFileURL, with: parameters, frameCount: loopFrameCount, progress: progress)
+            FileLog.shared.addMessage("VideoExporter Initial Video Ended: \(start.timeIntervalSinceNow)")
+
+            // Create & export final composition at full length by concatenating seconds long video from above until duration
+            let composition = try await createFinalComposition(from: temporaryFileURL, with: parameters, to: outputURL, progress: progress)
+            try await exportFinalComposition(composition, to: outputURL, duration: parameters.duration, fileType: parameters.fileType, progress: progress)
+
+            // Clean up temporary file
+            try? FileManager.default.removeItem(at: temporaryFileURL)
+            progress.completedUnitCount = progress.totalUnitCount
+            FileLog.shared.addMessage("VideoExporter Ended: \(start.timeIntervalSinceNow)")
+        } onCancel: {
+            progress.cancel()
+        }
+    }
+
+    //MARK: - Private Methods
+
+    // Step 1 of video export
+    private static func exportInitialVideo<Content: AnimatableContent>(of view: Content, to outputURL: URL, with parameters: Parameters, frameCount: Int, progress: Progress) async throws {
```

Comment:
This just creates the AVAsset related objects for exporting and then kicks off the `writeFrames` process.

Reply:
Could this GH comment be part of of the function/code comment for the method?